itest/lnd_route_blinding_test.go

```diff
@@ -1461,3 +1461,105 @@ func testBlindedPaymentHTLCReForward(ht *lntest.HarnessTest) {
+//   - Assert the error when attempting to create a blinded payment with an
+//     invalid path.
+//   - Create a blinded payment path when the blinded path is partially
+//     pre-specified.
+func testRestrictedBlindedRouteInvoices(ht *lntest.HarnessTest) {
+	// Create a five hop network: Alice -> Bob -> Carol -> Dave -> Erin.
+	chanAmt := btcutil.Amount(100000)
+	cfgs := [][]string{nil, nil, nil, nil, nil}
+	chanPoints, nodes := ht.CreateSimpleNetwork(
+		cfgs, lntest.OpenChannelParams{Amt: chanAmt},
+	)
+	alice, bob, carol, dave, erin := nodes[0], nodes[1], nodes[2], nodes[3],
+		nodes[4]
+	chanPointAliceBob, chanPointBobCarol, chanPointCarolDave,
+		chanPointDaveErin := chanPoints[0], chanPoints[1],
+		chanPoints[2], chanPoints[3]
+
+	// Lookup full channel info so that we have channel ids for our route.
+	aliceBobChan := ht.GetChannelByChanPoint(alice, chanPointAliceBob)
+	bobCarolChan := ht.GetChannelByChanPoint(bob, chanPointBobCarol)
+
+	// Wait for all nodes to have seen all channels.
+	for _, chanPoint := range chanPoints {
+		for _, node := range nodes {
+			ht.AssertChannelInGraph(node, chanPoint)
+		}
+	}
+
+	// 1) Let carol choose the wrong incoming channel.
+	var (
+		minNumRealHops      uint32 = 1
+		numHops             uint32 = 1
+		incomingChannelList        = []uint64{aliceBobChan.ChanId}
+	)
+
+	err := carol.RPC.AddInvoiceAssertErr(&lnrpc.Invoice{
+		Memo:      "test",
+		ValueMsat: 10_000_000,
+		IsBlinded: true,
+		BlindedPathConfig: &lnrpc.BlindedPathConfig{
+			MinNumRealHops:      &minNumRealHops,
+			NumHops:             &numHops,
+			IncomingChannelList: incomingChannelList,
+		},
+	})
+
+	errChan := fmt.Errorf("rpc error: code = Unknown desc = required"+
+		" channel %v at position %d not found",
+		aliceBobChan.ChanId, 1)
+	require.Equal(ht, err.Error(), errChan.Error())
+
+	// 2) Let carol set no income channel restrictions.
+	invoice := carol.RPC.AddInvoice(&lnrpc.Invoice{
+		Memo:      "test",
+		ValueMsat: 10_000_000,
+		IsBlinded: true,
+		BlindedPathConfig: &lnrpc.BlindedPathConfig{
+			MinNumRealHops: &minNumRealHops,
+			NumHops:        &numHops,
+		},
+	})
+
+	// Assert that it contains two blinded path with only 2 hops each one.
+	payReq := carol.RPC.DecodePayReq(invoice.PaymentRequest)
+	require.Len(ht, payReq.BlindedPaths, 2)
+	path := payReq.BlindedPaths[0].BlindedPath
+	require.Len(ht, path.BlindedHops, 2)
+	path = payReq.BlindedPaths[1].BlindedPath
+	require.Len(ht, path.BlindedHops, 2)
+
+	// 3) Let carol restrict bob as incoming channel.
+	incomingChannelList = []uint64{bobCarolChan.ChanId}
+
+	invoice = carol.RPC.AddInvoice(&lnrpc.Invoice{
+		Memo:      "test",
+		ValueMsat: 10_000_000,
+		IsBlinded: true,
+		BlindedPathConfig: &lnrpc.BlindedPathConfig{
+			MinNumRealHops:      &minNumRealHops,
+			NumHops:             &numHops,
+			IncomingChannelList: incomingChannelList,
+		},
+	})
+
+	// Assert that it contains a single blinded path with only
+	// 2 hops, with bob as the introduction node.
```

Comment:
i think it is worth adding one more test here (and it would require extending the network by 1 more node) 
just to ensure that the logic correctly builds on the specified list. 

Ie, an example where if the specified list is [A] and the max num hops is 2, then both `[A]` and [A, B]` should be chosen